llvm/lib/Analysis/DXILResource.cpp

```diff
@@ -770,6 +770,50 @@ void DXILBindingMap::print(raw_ostream &OS, DXILResourceTypeMap &DRTM,
   }
 }
 
+SmallVector<dxil::ResourceBindingInfo>
+DXILBindingMap::findByUse(const Value *Key) const {
+  const PHINode *Phi = dyn_cast<PHINode>(Key);
+  if (Phi) {
+    SmallVector<dxil::ResourceBindingInfo> Children;
+    for (const Value *V : Phi->operands()) {
+      Children.append(findByUse(V));
+    }
+    return Children;
+  }
+
+  const CallInst *CI = dyn_cast<CallInst>(Key);
+  if (!CI) {
+    return {};
+  }
+
+  const Type *UseType = CI->getType();
+
+  switch (CI->getIntrinsicID()) {
+  // Check if any of the parameters are the resource we are following. If so
+  // keep searching
+  case Intrinsic::not_intrinsic: {
+    SmallVector<dxil::ResourceBindingInfo> Children;
+    for (const Value *V : CI->args()) {
+      if (V->getType() != UseType) {
+        continue;
+      }
+
+      Children.append(findByUse(V));
+    }
+
+    return Children;
+  }
```

Comment:
Do we want to only do this for non-intrinsic calls, or should we do this for any call that isn't one of our resource handle intrinsics? I suspect we want a `default` label that just does `break` here and then to do handling of all calls outside of this switch.